pkg/hs/urbit-king/lib/Urbit/Vere/Pier.hs

```diff
@@ -122,17 +124,25 @@ runSerf
   -> RAcquire e Serf
 runSerf vSlog pax = do
   env <- ask
-  Serf.withSerf (config env)
+  serfProg <- io getSerfProg
+  Serf.withSerf (config env serfProg)
  where
   slog txt = atomically (readTVar vSlog) >>= (\f -> f txt)
-  config env = Serf.Config
-    { scSerf = env ^. pierConfigL . pcSerfExe . to unpack
+  config env serfProg = Serf.Config
+    { scSerf = env ^. pierConfigL . pcSerfExe . to (maybe serfProg unpack)
     , scPier = pax
     , scFlag = env ^. pierConfigL . pcSerfFlags
     , scSlog = \(pri, tank) -> printTank slog pri tank
     , scStdr = \txt -> slog (txt <> "\r\n")
     , scDead = pure () -- TODO: What can be done?
     }
+  getSerfProg :: IO FilePath
+  getSerfProg = do
+    (path, filename) <- splitFileName <$> getExecutablePath
+    pure $ case filename of
+      "urbit"      -> path </> "urbit-worker"
+      "urbit-king" -> path </> "urbit-worker"
+      _            -> "urbit-worker"
```

Comment:
The current vere logic is `concat(argv[0], "-worker")`. If we end up running in king-haskell in a similar manner to the current CI debug mode (wherein the binaries are prefixed with `debug-` or something), we'll need to revisit this logic.

I'm not sure we ever want to fallback to looking up `urbit-worker` in `$PATH`. But that code is not going to run in any current deployment scenario, so we can work out that behavior later.